roxie/ccd/ccdstate.cpp

```diff
@@ -713,7 +713,9 @@ class CRoxiePackageNode : extends CPackageNode, implements IRoxiePackage
             throw MakeStringException(ROXIE_FILE_ERROR, "Cannot write %s", fileName.str());
         // filename by now may be a local filename, or a dali one
         Owned<IRoxieDaliHelper> daliHelper = connectToDali();
-        Owned<ILocalOrDistributedFile> ldFile = createLocalOrDistributedFile(fileName, NULL, false, !resolveLocally(), true);
+        bool onlyLocal = fileNameServiceDali.isEmpty();
+        bool onlyDFS = !resolveLocally() && !onlyLocal;
```

Comment:
Looks good.  Although I'm a bit curious whether the !onlyLocal is necessary, since I thought resolveLocally() would return true in that case.

From resolveLocally(): " // default is false for explicit package files, but true for the default empty package".

And wouldn't not having a dali imply the default empty package?